capstone-app/src/app/modules/maps/map/map.component.spec.ts

```diff
@@ -136,6 +138,82 @@ describe('MapComponent', () => {
 
     expect(searchMarkers).toEqual(coordinates);
   });
+
+  fit('test geocoder', () => {
+    const fixture = TestBed.createComponent(MapComponent);
+    let coordinates: Coordinates = {
+      latitude: 26.011761,
+      longitude: -80.139053,
+      accuracy: undefined,
+      altitude: undefined,
+      altitudeAccuracy: undefined, 
+      heading: undefined,
+      speed: undefined,
+    }
+    let pos: Position = {
+      coords: coordinates,
+      timestamp: undefined,
+    }
+    spyOn(navigator.geolocation, 'getCurrentPosition').and.callFake(callback => {
+      fixture.componentInstance.locationCallbackSuccess(pos);
+    });
+    const addresses = [
+      '1405 N 12th Ct, Hollywood, FL 33019, USA', 
+      '1402 N 12th Ct, Hollywood, FL 33019, USA',
+      '1598-1300 N 12th Ct, Hollywood, FL 33019, USA',
+      'Hollywood Lakes, Hollywood, FL, USA',
+      'Hollywood, FL 33019, USA',
+      'Hollywood, FL, USA',
+      'Broward County, FL, USA',
+      'Florida, USA',
+      'United States'
+    ];
+    const searchLocation = new google.maps.LatLng(26.011761, -80.139053);
+    // let status: google.maps.GeocoderStatus;
+    spyOn(fixture.componentInstance.geocoder, 'geocode').and.callFake(({location: searchLocation}, callback) => {
+      let status: google.maps.GeocoderStatus = 'OK' as google.maps.GeocoderStatus;
+      const results: google.maps.GeocoderResult[] = addresses.map((address, index) => {
+        return {
+          formatted_address: address,
+          types: index === 5 ? ['locality'] : [],
```

Comment:
nit: Forgot to mention, but you might consider adding a comment specifying why we check for index === 5 (i.e. in order to add the locality type to the geocoder result corresponding to the 'Hollywood, FL, USA' address)